source/slang/slang.cpp

```diff
@@ -1631,7 +1663,14 @@ void TranslationUnitRequest::_addSourceFile(SourceFile* sourceFile)
         // for non-file-based dependencies later when shader files are being hashed for
         // the shader cache.
 
-        // TODO: Add a file path dependency using a fake path string for this source
+        slang::Checksum sourceHash;
+        MD5Context context;
+        MD5HashGen hashGen;
```

Comment:
These five lines should just become another helper function.
How about `slang::Checksum computeChecksumForString(UnownedStringSlice text)`?

Reply:
I already made an `update` helper for UnownedStringSlice, and it might look a bit weird to turn all five of these lines into a single helper when none of the other hashes are structured like that

Reply:
We could have a lot of other places where we just want to compute MD5 for a string and nothing else, so having this helper will be useful in many ways.